lib/connection/result/row_stream.js

```diff
@@ -428,17 +437,27 @@ function buildMapColumnExtractFnNames(columns, fetchAsString)
  * @param {Object} row
  * @param {Object[]} columns
  * @param {Object} [mapColumnIdToExtractFnName]
+ * @param {String?} rowMode - "array" or "object", default is "object"
  *
  * @returns {Object}
  */
-function externalizeRow(row, columns, mapColumnIdToExtractFnName)
+function externalizeRow(row, columns, mapColumnIdToExtractFnName, rowMode, duplicatedColumnNames)
 {
-  var externalizedRow = {};
+  const isArrayRowMode = (rowMode) => {
```

Comment:
Do we need a function here? Maybe it can be inlined? `rowMode == 'array'` seems to be readable enough.